{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from pymodulon.io import *\n",
    "from pymodulon.motif import *"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "ica_data = load_json_model(os.path.join('..','data','processed_data','bd_raw.json.gz'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "fasta_file = os.path.join('..','data','external','genome.fasta')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Search for motifs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import re\n",
    "import subprocess\n",
    "from typing import Dict, List, Optional, Union\n",
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "from bs4 import BeautifulSoup\n",
    "\n",
    "import pymodulon.data.meme\n",
    "from pymodulon.core import IcaData, MotifInfo"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Finding motifs for 43 sequences\n",
      "Running command: meme motifs/Unchar-1.fasta -oc motifs/Unchar-1 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 14\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-1'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 43 p0 43 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 25800 highwater mark: seq 43 pos 594\n",
      "BALANCE: samples 43 chars 25800 nodes 8 chars/node 3225\n",
      "Initializing the motif probability tables for 2 to 43 sites...\n",
      "nsites = 40\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    43, min_w= 600, max_w=  600, total_size=    25800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 14 28 43\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  43, iter=  30 \n",
      "motif=2\n",
      "SEED DEPTHS: 14 28 43\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  43, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 14 28 43\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  43, iter=  20 \n",
      "motif=4\n",
      "SEED DEPTHS: 14 28 43\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  43, iter=  20 \n",
      "motif=5\n",
      "SEED DEPTHS: 14 28 43\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  43, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 157 sites\n",
      "Finding motifs for 29 sequences\n",
      "Running command: meme motifs/T3SS.fasta -oc motifs/T3SS -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/T3SS'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 29 p0 29 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 17400 highwater mark: seq 29 pos 594\n",
      "BALANCE: samples 29 chars 17400 nodes 8 chars/node 2175\n",
      "Initializing the motif probability tables for 2 to 29 sites...\n",
      "nsites = 24\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    29, min_w= 600, max_w=  600, total_size=    17400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 9 18 29\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  29, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 9 18 29\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  29, iter=  10 \n",
      "motif=3\n",
      "SEED DEPTHS: 9 18 29\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  29, iter=  10 \n",
      "motif=4\n",
      "SEED DEPTHS: 9 18 29\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  29, iter=  10 \n",
      "motif=5\n",
      "SEED DEPTHS: 9 18 29\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  29, iter=  20 \n",
      "motif=6\n",
      "SEED DEPTHS: 9 18 29\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  29, iter=  10 \n",
      "motif=7\n",
      "SEED DEPTHS: 9 18 29\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  29, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 6 motifs across 54 sites\n",
      "Finding motifs for 13 sequences\n",
      "Running command: meme motifs/Unchar-2.fasta -oc motifs/Unchar-2 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-2'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 13 p0 13 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 7800 highwater mark: seq 13 pos 594\n",
      "BALANCE: samples 13 chars 7800 nodes 8 chars/node 975\n",
      "Initializing the motif probability tables for 2 to 13 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    13, min_w= 600, max_w=  600, total_size=     7800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 4 8 13\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  13, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/FixK2-1'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 84 p0 84 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 50400 highwater mark: seq 84 pos 594\n",
      "BALANCE: samples 84 chars 50400 nodes 8 chars/node 6300\n",
      "Initializing the motif probability tables for 2 to 84 sites...\n",
      "nsites = 72\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 84 sequences\n",
      "Running command: meme motifs/FixK2-1.fasta -oc motifs/FixK2-1 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nsites = 80\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    84, min_w= 600, max_w=  600, total_size=    50400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 28 56 84\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  84, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 28 56 84\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  84, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 28 56 84\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  84, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 28 56 84\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  84, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 28 56 84\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  84, iter=  40 \n",
      "motif=6\n",
      "SEED DEPTHS: 28 56 84\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  84, iter=  40 \n",
      "motif=7\n",
      "SEED DEPTHS: 28 56 84\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  84, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 6 motifs across 362 sites\n",
      "Finding motifs for 41 sequences\n",
      "Running command: meme motifs/bll2166.fasta -oc motifs/bll2166 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/bll2166'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 41 p0 41 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 24600 highwater mark: seq 41 pos 594\n",
      "BALANCE: samples 41 chars 24600 nodes 8 chars/node 3075\n",
      "Initializing the motif probability tables for 2 to 41 sites...\n",
      "nsites = 40\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    41, min_w= 600, max_w=  600, total_size=    24600\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  20 \n",
      "motif=2\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  30 \n",
      "motif=3\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  20 \n",
      "motif=4\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 117 sites\n",
      "Finding motifs for 88 sequences\n",
      "Running command: meme motifs/NifA.fasta -oc motifs/NifA -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/NifA'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 88 p0 88 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 52800 highwater mark: seq 88 pos 594\n",
      "BALANCE: samples 88 chars 52800 nodes 8 chars/node 6600\n",
      "Initializing the motif probability tables for 2 to 88 sites...\n",
      "nsites = 88\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    88, min_w= 600, max_w=  600, total_size=    52800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 29 58 88\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  88, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 29 58 88\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  88, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 29 58 88\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  88, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 29 58 88\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  88, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 29 58 88\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  88, iter=  40 \n",
      "motif=6\n",
      "SEED DEPTHS: 29 58 88\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  88, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 5 motifs across 220 sites\n",
      "Finding motifs for 85 sequences\n",
      "Running command: meme motifs/PhyR-1.fasta -oc motifs/PhyR-1 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 28\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/PhyR-1'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 85 p0 85 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 51000 highwater mark: seq 85 pos 594\n",
      "BALANCE: samples 85 chars 51000 nodes 8 chars/node 6375\n",
      "Initializing the motif probability tables for 2 to 85 sites...\n",
      "nsites = 80\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    85, min_w= 600, max_w=  600, total_size=    51000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 28 56 85\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  85, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 28 56 85\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  85, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 28 56 85\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  85, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 28 56 85\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  85, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 28 56 85\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  85, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 253 sites\n",
      "Finding motifs for 9 sequences\n",
      "Running command: meme motifs/Denitrification.fasta -oc motifs/Denitrification -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Denitrification'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 9 p0 9 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 5400 highwater mark: seq 9 pos 594\n",
      "BALANCE: samples 9 chars 5400 nodes 8 chars/node 675\n",
      "Initializing the motif probability tables for 2 to 9 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     9, min_w= 600, max_w=  600, total_size=     5400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 9\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   9, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 3 6 9\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   9, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 3 6 9\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   9, iter=   0 \n",
      "motif=4\n",
      "SEED DEPTHS: 3 6 9\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   9, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 12 sites\n",
      "Less than two sequences found for iModulon: Null-1\n",
      "Finding motifs for 9 sequences\n",
      "Running command: meme motifs/Methane.fasta -oc motifs/Methane -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Methane'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 9 p0 9 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 5400 highwater mark: seq 9 pos 594\n",
      "BALANCE: samples 9 chars 5400 nodes 8 chars/node 675\n",
      "Initializing the motif probability tables for 2 to 9 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     9, min_w= 600, max_w=  600, total_size=     5400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 9\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   9, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Unchar-3'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 10 p0 10 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 6000 highwater mark: seq 10 pos 594\n",
      "BALANCE: samples 10 chars 6000 nodes 8 chars/node 750\n",
      "Initializing the motif probability tables for 2 to 10 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    10, min_w= 600, max_w=  600, total_size=     6000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=9, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 10 sequences\n",
      "Running command: meme motifs/Unchar-3.fasta -oc motifs/Unchar-3 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  10, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Irr'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 34 p0 34 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 20400 highwater mark: seq 34 pos 594\n",
      "BALANCE: samples 34 chars 20400 nodes 8 chars/node 2550\n",
      "Initializing the motif probability tables for 2 to 34 sites...\n",
      "nsites = 32\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    34, min_w= 600, max_w=  600, total_size=    20400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 11 22 34\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 34 sequences\n",
      "Running command: meme motifs/Irr.fasta -oc motifs/Irr -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  34, iter=  20 \n",
      "motif=2\n",
      "SEED DEPTHS: 11 22 34\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  34, iter=  30 \n",
      "motif=3\n",
      "SEED DEPTHS: 11 22 34\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  34, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 11 22 34\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  34, iter=  30 \n",
      "motif=5\n",
      "SEED DEPTHS: 11 22 34\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  34, iter=  30 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 92 sites\n",
      "Finding motifs for 32 sequences\n",
      "Running command: meme motifs/Multiple_stress.fasta -oc motifs/Multiple_stress -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 10\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Multiple_stress'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 32 p0 32 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 19200 highwater mark: seq 32 pos 594\n",
      "BALANCE: samples 32 chars 19200 nodes 8 chars/node 2400\n",
      "Initializing the motif probability tables for 2 to 32 sites...\n",
      "nsites = 32\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    32, min_w= 600, max_w=  600, total_size=    19200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  10 \n",
      "motif=2\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  20 \n",
      "motif=3\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  10 \n",
      "motif=4\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  10 \n",
      "motif=5\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  10 \n",
      "motif=6\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  10 \n",
      "motif=7\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  10 \n",
      "motif=8\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  20 \n",
      "motif=9\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  10 \n",
      "motif=10\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  20 \n",
      "motif=11\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  30 \n",
      "motif=12\n",
      "SEED DEPTHS: 10 20 32\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  32, iter=  20 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 11 motifs across 161 sites\n",
      "Finding motifs for 5 sequences\n",
      "Running command: meme motifs/bll3348.fasta -oc motifs/bll3348 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/bll3348'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 5 p0 5 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 3000 highwater mark: seq 5 pos 594\n",
      "BALANCE: samples 5 chars 3000 nodes 8 chars/node 375\n",
      "Initializing the motif probability tables for 2 to 5 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     5, min_w= 600, max_w=  600, total_size=     3000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2 4 5\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   5, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 2 4 5\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   5, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 2 4 5\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   5, iter=   0 \n",
      "motif=4\n",
      "SEED DEPTHS: 2 4 5\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   5, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/Unchar-4'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 2 p0 2 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 1200 highwater mark: seq 2 pos 594\n",
      "BALANCE: samples 2 chars 1200 nodes 8 chars/node 150\n",
      "Initializing the motif probability tables for 2 to 2 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     2, min_w= 600, max_w=  600, total_size=     1200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=19, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 10 sites\n",
      "Finding motifs for 2 sequences\n",
      "Running command: meme motifs/Unchar-4.fasta -oc motifs/Unchar-4 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "starts: w=20, seq=0, l=600\t\t\t\t\r",
      "starts: w=21, seq=0, l=600\t\t\t\t\r",
      "starts: w=22, seq=0, l=600\t\t\t\t\r",
      "starts: w=23, seq=0, l=600\t\t\t\t\r",
      "starts: w=24, seq=0, l=600\t\t\t\t\r",
      "starts: w=25, seq=0, l=600\t\t\t\t\r",
      "starts: w=26, seq=0, l=600\t\t\t\t\r",
      "starts: w=27, seq=0, l=600\t\t\t\t\r",
      "starts: w=28, seq=0, l=600\t\t\t\t\r",
      "starts: w=29, seq=0, l=600\t\t\t\t\r",
      "starts: w=30, seq=0, l=600\t\t\t\t\r",
      "starts: w=31, seq=0, l=600\t\t\t\t\r",
      "starts: w=32, seq=0, l=600\t\t\t\t\r",
      "starts: w=33, seq=0, l=600\t\t\t\t\r",
      "starts: w=34, seq=0, l=600\t\t\t\t\r",
      "starts: w=35, seq=0, l=600\t\t\t\t\r",
      "starts: w=36, seq=0, l=600\t\t\t\t\r",
      "starts: w=37, seq=0, l=600\t\t\t\t\r",
      "starts: w=38, seq=0, l=600\t\t\t\t\r",
      "starts: w=39, seq=0, l=600\t\t\t\t\r",
      "starts: w=40, seq=0, l=600\t\t\t\t\r",
      "\r",
      "em: w=   6, psites=   2, iter=   0 \r",
      "em: w=   6, psites=   2, iter=  10 \r",
      "em: w=   6, psites=   2, iter=  20 \r",
      "em: w=   6, psites=   2, iter=  30 \r",
      "em: w=  14, psites=   2, iter=   0 \r",
      "em: w=  22, psites=   2, iter=   0 \r",
      "em: w=  30, psites=   2, iter=   0 \r",
      "em: w=  38, psites=   2, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Less than two sequences found for iModulon: Genomic-1\n",
      "Finding motifs for 18 sequences\n",
      "Running command: meme motifs/Unchar-5.fasta -oc motifs/Unchar-5 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-5'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 18 p0 18 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 10800 highwater mark: seq 18 pos 594\n",
      "BALANCE: samples 18 chars 10800 nodes 8 chars/node 1350\n",
      "Initializing the motif probability tables for 2 to 18 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    18, min_w= 600, max_w=  600, total_size=    10800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/XylR_bll3916'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 154 p0 154 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 60303 highwater mark: seq 102 pos 202\n",
      "BALANCE: samples 102 chars 61200 nodes 8 chars/node 7650\n",
      "Initializing the motif probability tables for 2 to 154 sites...\n",
      "nsites = 152\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 154 sequences\n",
      "Running command: meme motifs/XylR_bll3916.fasta -oc motifs/XylR_bll3916 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 51\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=   154, min_w= 600, max_w=  600, total_size=    92400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 51 102 154\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 154, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 51 102 154\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 154, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 51 102 154\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 154, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 51 102 154\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 154, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 51 102 154\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 154, iter=  40 \n",
      "motif=6\n",
      "SEED DEPTHS: 51 102 154\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 154, iter=  40 \n",
      "motif=7\n",
      "SEED DEPTHS: 51 102 154\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 154, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 6 motifs across 655 sites\n",
      "Finding motifs for 2 sequences\n",
      "Running command: meme motifs/Genomic-2.fasta -oc motifs/Genomic-2 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Genomic-2'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 2 p0 2 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 1200 highwater mark: seq 2 pos 594\n",
      "BALANCE: samples 2 chars 1200 nodes 8 chars/node 150\n",
      "Initializing the motif probability tables for 2 to 2 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     2, min_w= 600, max_w=  600, total_size=     1200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  38, psites=   2, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Unchar-6'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 25 p0 25 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 15000 highwater mark: seq 25 pos 594\n",
      "BALANCE: samples 25 chars 15000 nodes 8 chars/node 1875\n",
      "Initializing the motif probability tables for 2 to 25 sites...\n",
      "nsites = 24\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    25, min_w= 600, max_w=  600, total_size=    15000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 25 sequences\n",
      "Running command: meme motifs/Unchar-6.fasta -oc motifs/Unchar-6 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  25, iter=  10 \n",
      "motif=2\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  25, iter=  10 \n",
      "motif=3\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  25, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2 motifs across 50 sites\n",
      "Finding motifs for 87 sequences\n",
      "Running command: meme motifs/Unchar-7.fasta -oc motifs/Unchar-7 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 29\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-7'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 87 p0 87 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 52200 highwater mark: seq 87 pos 594\n",
      "BALANCE: samples 87 chars 52200 nodes 8 chars/node 6525\n",
      "Initializing the motif probability tables for 2 to 87 sites...\n",
      "nsites = 80\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    87, min_w= 600, max_w=  600, total_size=    52200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 29 58 87\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  87, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 29 58 87\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  87, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 29 58 87\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  87, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 29 58 87\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  87, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 29 58 87\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  87, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 299 sites\n",
      "Finding motifs for 103 sequences\n",
      "Running command: meme motifs/HrcA.fasta -oc motifs/HrcA -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 34\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/HrcA'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 103 p0 103 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 60248 highwater mark: seq 102 pos 147\n",
      "BALANCE: samples 102 chars 61200 nodes 8 chars/node 7650\n",
      "Initializing the motif probability tables for 2 to 103 sites...\n",
      "nsites = 96\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=   103, min_w= 600, max_w=  600, total_size=    61800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 34 68 103\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 103, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 34 68 103\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 103, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 34 68 103\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 103, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 34 68 103\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 103, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 34 68 103\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 103, iter=  40 \n",
      "motif=6\n",
      "SEED DEPTHS: 34 68 103\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 103, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 5 motifs across 370 sites\n",
      "Finding motifs for 19 sequences\n",
      "Running command: meme motifs/Signal_transduction.fasta -oc motifs/Signal_transduction -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Signal_transduction'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 19 p0 19 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 11400 highwater mark: seq 19 pos 594\n",
      "BALANCE: samples 19 chars 11400 nodes 8 chars/node 1425\n",
      "Initializing the motif probability tables for 2 to 19 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    19, min_w= 600, max_w=  600, total_size=    11400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/Leucine'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 16 p0 16 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 9600 highwater mark: seq 16 pos 594\n",
      "BALANCE: samples 16 chars 9600 nodes 8 chars/node 1200\n",
      "Initializing the motif probability tables for 2 to 16 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    16, min_w= 600, max_w=  600, total_size=     9600\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 5 10 16\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=7, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 1 motif across 9 sites\n",
      "Finding motifs for 16 sequences\n",
      "Running command: meme motifs/Leucine.fasta -oc motifs/Leucine -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  16, iter=  10 \n",
      "motif=2\n",
      "SEED DEPTHS: 5 10 16\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  16, iter=  10 \n",
      "motif=3\n",
      "SEED DEPTHS: 5 10 16\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  16, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/Flagella'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 19 p0 19 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 11400 highwater mark: seq 19 pos 594\n",
      "BALANCE: samples 19 chars 11400 nodes 8 chars/node 1425\n",
      "Initializing the motif probability tables for 2 to 19 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    19, min_w= 600, max_w=  600, total_size=    11400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2 motifs across 18 sites\n",
      "Finding motifs for 19 sequences\n",
      "Running command: meme motifs/Flagella.fasta -oc motifs/Flagella -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  19, iter=  10 \n",
      "motif=2\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=   0 \n",
      "motif=4\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=   0 \n",
      "motif=5\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=  10 \n",
      "motif=6\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 5 motifs across 73 sites\n",
      "Finding motifs for 41 sequences\n",
      "Running command: meme motifs/Translation-1.fasta -oc motifs/Translation-1 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Translation-1'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 41 p0 41 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 24600 highwater mark: seq 41 pos 594\n",
      "BALANCE: samples 41 chars 24600 nodes 8 chars/node 3075\n",
      "Initializing the motif probability tables for 2 to 41 sites...\n",
      "nsites = 40\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    41, min_w= 600, max_w=  600, total_size=    24600\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  30 \n",
      "motif=2\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  30 \n",
      "motif=4\n",
      "SEED DEPTHS: 13 26 41\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  41, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 96 sites\n",
      "Finding motifs for 2 sequences\n",
      "Running command: meme motifs/Calvin_cycle.fasta -oc motifs/Calvin_cycle -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Calvin_cycle'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 2 p0 2 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 1200 highwater mark: seq 2 pos 594\n",
      "BALANCE: samples 2 chars 1200 nodes 8 chars/node 150\n",
      "Initializing the motif probability tables for 2 to 2 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     2, min_w= 600, max_w=  600, total_size=     1200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  38, psites=   2, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Unchar-8'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 40 p0 40 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 24000 highwater mark: seq 40 pos 594\n",
      "BALANCE: samples 40 chars 24000 nodes 8 chars/node 3000\n",
      "Initializing the motif probability tables for 2 to 40 sites...\n",
      "nsites = 40\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    40, min_w= 600, max_w=  600, total_size=    24000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 40 sequences\n",
      "Running command: meme motifs/Unchar-8.fasta -oc motifs/Unchar-8 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  40, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  40, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  40, iter=  30 \n",
      "motif=4\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  40, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 120 sites\n",
      "Finding motifs for 17 sequences\n",
      "Running command: meme motifs/Oxidation_stress.fasta -oc motifs/Oxidation_stress -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Oxidation_stress'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 17 p0 17 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 10200 highwater mark: seq 17 pos 594\n",
      "BALANCE: samples 17 chars 10200 nodes 8 chars/node 1275\n",
      "Initializing the motif probability tables for 2 to 17 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    17, min_w= 600, max_w=  600, total_size=    10200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 5 10 17\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  17, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 5 10 17\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  17, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/Unchar-9'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 40 p0 40 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 24000 highwater mark: seq 40 pos 594\n",
      "BALANCE: samples 40 chars 24000 nodes 8 chars/node 3000\n",
      "Initializing the motif probability tables for 2 to 40 sites...\n",
      "nsites = 40\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    40, min_w= 600, max_w=  600, total_size=    24000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 1 motif across 17 sites\n",
      "Finding motifs for 40 sequences\n",
      "Running command: meme motifs/Unchar-9.fasta -oc motifs/Unchar-9 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 13\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  40, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  40, iter=  20 \n",
      "motif=3\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  40, iter=  30 \n",
      "motif=4\n",
      "SEED DEPTHS: 13 26 40\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  40, iter=  20 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 120 sites\n",
      "Finding motifs for 16 sequences\n",
      "Running command: meme motifs/Phenylalanine.fasta -oc motifs/Phenylalanine -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Phenylalanine'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 16 p0 16 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 9600 highwater mark: seq 16 pos 594\n",
      "BALANCE: samples 16 chars 9600 nodes 8 chars/node 1200\n",
      "Initializing the motif probability tables for 2 to 16 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    16, min_w= 600, max_w=  600, total_size=     9600\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 5 10 16\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  16, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 5 10 16\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  16, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 5 10 16\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  16, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/FixK2-2'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 122 p0 122 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 60270 highwater mark: seq 102 pos 169\n",
      "BALANCE: samples 102 chars 61200 nodes 8 chars/node 7650\n",
      "Initializing the motif probability tables for 2 to 122 sites...\n",
      "nsites = 64\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2 motifs across 31 sites\n",
      "Finding motifs for 122 sequences\n",
      "Running command: meme motifs/FixK2-2.fasta -oc motifs/FixK2-2 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 40\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "nsites = 120\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=   122, min_w= 600, max_w=  600, total_size=    73200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 40 80 122\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 122, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 40 80 122\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 122, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 40 80 122\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 122, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 40 80 122\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 122, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 40 80 122\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 122, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 417 sites\n",
      "Finding motifs for 18 sequences\n",
      "Running command: meme motifs/Unchar-10.fasta -oc motifs/Unchar-10 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-10'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 18 p0 18 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 10800 highwater mark: seq 18 pos 594\n",
      "BALANCE: samples 18 chars 10800 nodes 8 chars/node 1350\n",
      "Initializing the motif probability tables for 2 to 18 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    18, min_w= 600, max_w=  600, total_size=    10800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/Unchar-11'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 24 p0 24 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 14400 highwater mark: seq 24 pos 594\n",
      "BALANCE: samples 24 chars 14400 nodes 8 chars/node 1800\n",
      "Initializing the motif probability tables for 2 to 24 sites...\n",
      "nsites = 24\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    24, min_w= 600, max_w=  600, total_size=    14400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 8 16 24\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 1 motif across 14 sites\n",
      "Finding motifs for 24 sequences\n",
      "Running command: meme motifs/Unchar-11.fasta -oc motifs/Unchar-11 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  24, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 8 16 24\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  24, iter=  10 \n",
      "motif=3\n",
      "SEED DEPTHS: 8 16 24\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  24, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2 motifs across 45 sites\n",
      "Finding motifs for 3 sequences\n",
      "Running command: meme motifs/Translation-2.fasta -oc motifs/Translation-2 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Translation-2'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 3 p0 3 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 1800 highwater mark: seq 3 pos 594\n",
      "BALANCE: samples 3 chars 1800 nodes 8 chars/node 225\n",
      "Initializing the motif probability tables for 2 to 3 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     3, min_w= 600, max_w=  600, total_size=     1800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2 3\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  38, psites=   2, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/HutC'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 4 p0 4 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 2400 highwater mark: seq 4 pos 594\n",
      "BALANCE: samples 4 chars 2400 nodes 8 chars/node 300\n",
      "Initializing the motif probability tables for 2 to 4 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     4, min_w= 600, max_w=  600, total_size=     2400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2 4\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=13, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Less than two sequences found for iModulon: Null-2\n",
      "Finding motifs for 4 sequences\n",
      "Running command: meme motifs/HutC.fasta -oc motifs/HutC -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  38, psites=   2, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 2 4\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  38, psites=   2, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 2 4\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  38, psites=   2, iter=   0 \n",
      "motif=4\n",
      "SEED DEPTHS: 2 4\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  38, psites=   2, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/Genomic-3'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 6 p0 6 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 3600 highwater mark: seq 6 pos 594\n",
      "BALANCE: samples 6 chars 3600 nodes 8 chars/node 450\n",
      "Initializing the motif probability tables for 2 to 6 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     6, min_w= 600, max_w=  600, total_size=     3600\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2 4 6\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=8, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 10 sites\n",
      "Finding motifs for 6 sequences\n",
      "Running command: meme motifs/Genomic-3.fasta -oc motifs/Genomic-3 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=   6, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Unchar-12'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 19 p0 19 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 11400 highwater mark: seq 19 pos 594\n",
      "BALANCE: samples 19 chars 11400 nodes 8 chars/node 1425\n",
      "Initializing the motif probability tables for 2 to 19 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    19, min_w= 600, max_w=  600, total_size=    11400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=7, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 19 sequences\n",
      "Running command: meme motifs/Unchar-12.fasta -oc motifs/Unchar-12 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  19, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=  20 \n",
      "motif=4\n",
      "SEED DEPTHS: 6 12 19\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  19, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 29 sites\n",
      "Less than two sequences found for iModulon: Noise\n",
      "Finding motifs for 14 sequences\n",
      "Running command: meme motifs/Unchar-13.fasta -oc motifs/Unchar-13 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-13'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 14 p0 14 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 8400 highwater mark: seq 14 pos 594\n",
      "BALANCE: samples 14 chars 8400 nodes 8 chars/node 1050\n",
      "Initializing the motif probability tables for 2 to 14 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    14, min_w= 600, max_w=  600, total_size=     8400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 4 8 14\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  14, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Unchar-14'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 61 p0 61 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 36600 highwater mark: seq 61 pos 594\n",
      "BALANCE: samples 61 chars 36600 nodes 8 chars/node 4575\n",
      "Initializing the motif probability tables for 2 to 61 sites...\n",
      "nsites = 56\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    61, min_w= 600, max_w=  600, total_size=    36600\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 20 40 61\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 61 sequences\n",
      "Running command: meme motifs/Unchar-14.fasta -oc motifs/Unchar-14 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 20\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  61, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 20 40 61\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  61, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 20 40 61\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  61, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2 motifs across 113 sites\n",
      "Finding motifs for 12 sequences\n",
      "Running command: meme motifs/AA_transport.fasta -oc motifs/AA_transport -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/AA_transport'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 12 p0 12 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 7200 highwater mark: seq 12 pos 594\n",
      "BALANCE: samples 12 chars 7200 nodes 8 chars/node 900\n",
      "Initializing the motif probability tables for 2 to 12 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    12, min_w= 600, max_w=  600, total_size=     7200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 4 8 12\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  12, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 4 8 12\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  12, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 4 8 12\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  12, iter=   0 \n",
      "motif=4\n",
      "SEED DEPTHS: 4 8 12\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  12, iter=   0 \n",
      "motif=5\n",
      "SEED DEPTHS: 4 8 12\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  12, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 27 sites\n",
      "Finding motifs for 56 sequences\n",
      "Running command: meme motifs/Glyoxylate.fasta -oc motifs/Glyoxylate -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Glyoxylate'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 56 p0 56 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 33600 highwater mark: seq 56 pos 594\n",
      "BALANCE: samples 56 chars 33600 nodes 8 chars/node 4200\n",
      "Initializing the motif probability tables for 2 to 56 sites...\n",
      "nsites = 56\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    56, min_w= 600, max_w=  600, total_size=    33600\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 18 36 56\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  56, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 18 36 56\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  56, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 18 36 56\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  56, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 18 36 56\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  56, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 168 sites\n",
      "Finding motifs for 2 sequences\n",
      "Running command: meme motifs/Unchar-15.fasta -oc motifs/Unchar-15 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-15'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 2 p0 2 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 1200 highwater mark: seq 2 pos 594\n",
      "BALANCE: samples 2 chars 1200 nodes 8 chars/node 150\n",
      "Initializing the motif probability tables for 2 to 2 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     2, min_w= 600, max_w=  600, total_size=     1200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  38, psites=   2, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/NodD2'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 10 p0 10 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 6000 highwater mark: seq 10 pos 594\n",
      "BALANCE: samples 10 chars 6000 nodes 8 chars/node 750\n",
      "Initializing the motif probability tables for 2 to 10 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    10, min_w= 600, max_w=  600, total_size=     6000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=9, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 10 sequences\n",
      "Running command: meme motifs/NodD2.fasta -oc motifs/NodD2 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  10, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/NnrR'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 23 p0 23 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 13800 highwater mark: seq 23 pos 594\n",
      "BALANCE: samples 23 chars 13800 nodes 8 chars/node 1725\n",
      "Initializing the motif probability tables for 2 to 23 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    23, min_w= 600, max_w=  600, total_size=    13800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 7 14 23\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 23 sequences\n",
      "Running command: meme motifs/NnrR.fasta -oc motifs/NnrR -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 7\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  23, iter=  10 \n",
      "motif=2\n",
      "SEED DEPTHS: 7 14 23\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  23, iter=  10 \n",
      "motif=3\n",
      "SEED DEPTHS: 7 14 23\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  23, iter=  10 \n",
      "motif=4\n",
      "SEED DEPTHS: 7 14 23\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  23, iter=  20 \n",
      "motif=5\n",
      "SEED DEPTHS: 7 14 23\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  23, iter=  20 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 61 sites\n",
      "Finding motifs for 119 sequences\n",
      "Running command: meme motifs/Cell_mobility.fasta -oc motifs/Cell_mobility -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 39\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Cell_mobility'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 119 p0 119 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 60267 highwater mark: seq 102 pos 166\n",
      "BALANCE: samples 102 chars 61200 nodes 8 chars/node 7650\n",
      "Initializing the motif probability tables for 2 to 119 sites...\n",
      "nsites = 112\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=   119, min_w= 600, max_w=  600, total_size=    71400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 39 78 119\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 119, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 39 78 119\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 119, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 39 78 119\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 119, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 39 78 119\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 119, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 39 78 119\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 119, iter=  40 \n",
      "motif=6\n",
      "SEED DEPTHS: 39 78 119\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 119, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 5 motifs across 443 sites\n",
      "Finding motifs for 33 sequences\n",
      "Running command: meme motifs/T6SS.fasta -oc motifs/T6SS -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 11\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/T6SS'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 33 p0 33 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 19800 highwater mark: seq 33 pos 594\n",
      "BALANCE: samples 33 chars 19800 nodes 8 chars/node 2475\n",
      "Initializing the motif probability tables for 2 to 33 sites...\n",
      "nsites = 32\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    33, min_w= 600, max_w=  600, total_size=    19800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 11 22 33\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  33, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 11 22 33\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  33, iter=  30 \n",
      "motif=3\n",
      "SEED DEPTHS: 11 22 33\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  33, iter=  20 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "Writing results to output directory 'motifs/Sulfur-1'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 18 p0 18 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 10800 highwater mark: seq 18 pos 594\n",
      "BALANCE: samples 18 chars 10800 nodes 8 chars/node 1350\n",
      "Initializing the motif probability tables for 2 to 18 sites...\n",
      "nsites = 16\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    18, min_w= 600, max_w=  600, total_size=    10800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "starts: w=6, seq=0, l=600\t\t\t\t\r"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 2 motifs across 66 sites\n",
      "Finding motifs for 18 sequences\n",
      "Running command: meme motifs/Sulfur-1.fasta -oc motifs/Sulfur-1 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 6\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  40, psites=  18, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=  10 \n",
      "motif=4\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "motif=5\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "motif=6\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "motif=7\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "motif=8\n",
      "SEED DEPTHS: 6 12 18\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  18, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 7 motifs across 65 sites\n",
      "Finding motifs for 25 sequences\n",
      "Running command: meme motifs/Unchar-16.fasta -oc motifs/Unchar-16 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 8\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-16'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 25 p0 25 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 15000 highwater mark: seq 25 pos 594\n",
      "BALANCE: samples 25 chars 15000 nodes 8 chars/node 1875\n",
      "Initializing the motif probability tables for 2 to 25 sites...\n",
      "nsites = 24\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    25, min_w= 600, max_w=  600, total_size=    15000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  25, iter=  20 \n",
      "motif=2\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  25, iter=  10 \n",
      "motif=3\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  25, iter=  10 \n",
      "motif=4\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  25, iter=   0 \n",
      "motif=5\n",
      "SEED DEPTHS: 8 16 25\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  25, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 97 sites\n",
      "Finding motifs for 10 sequences\n",
      "Running command: meme motifs/Copper.fasta -oc motifs/Copper -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Copper'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 10 p0 10 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 6000 highwater mark: seq 10 pos 594\n",
      "BALANCE: samples 10 chars 6000 nodes 8 chars/node 750\n",
      "Initializing the motif probability tables for 2 to 10 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    10, min_w= 600, max_w=  600, total_size=     6000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=4\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=5\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 27 sites\n",
      "Finding motifs for 15 sequences\n",
      "Running command: meme motifs/Unchar-17.fasta -oc motifs/Unchar-17 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-17'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 15 p0 15 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 9000 highwater mark: seq 15 pos 594\n",
      "BALANCE: samples 15 chars 9000 nodes 8 chars/node 1125\n",
      "Initializing the motif probability tables for 2 to 15 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    15, min_w= 600, max_w=  600, total_size=     9000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 5 10 15\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  15, iter=  10 \n",
      "motif=2\n",
      "SEED DEPTHS: 5 10 15\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  15, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 1 motif across 14 sites\n",
      "Finding motifs for 130 sequences\n",
      "Running command: meme motifs/Bacterial_chemotaxis.fasta -oc motifs/Bacterial_chemotaxis -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 43\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Bacterial_chemotaxis'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 130 p0 130 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 60279 highwater mark: seq 102 pos 178\n",
      "BALANCE: samples 102 chars 61200 nodes 8 chars/node 7650\n",
      "Initializing the motif probability tables for 2 to 130 sites...\n",
      "nsites = 128\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=   130, min_w= 600, max_w=  600, total_size=    78000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 43 86 130\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 130, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 43 86 130\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 130, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 43 86 130\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 130, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 43 86 130\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 130, iter=  40 \n",
      "motif=5\n",
      "SEED DEPTHS: 43 86 130\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 130, iter=  40 \n",
      "motif=6\n",
      "SEED DEPTHS: 43 86 130\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 130, iter=  40 \n",
      "motif=7\n",
      "SEED DEPTHS: 43 86 130\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites= 130, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 6 motifs across 516 sites\n",
      "Finding motifs for 9 sequences\n",
      "Running command: meme motifs/BioR.fasta -oc motifs/BioR -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/BioR'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 9 p0 9 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 5400 highwater mark: seq 9 pos 594\n",
      "BALANCE: samples 9 chars 5400 nodes 8 chars/node 675\n",
      "Initializing the motif probability tables for 2 to 9 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     9, min_w= 600, max_w=  600, total_size=     5400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 9\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   9, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Less than two sequences found for iModulon: Null-3\n",
      "Finding motifs for 10 sequences\n",
      "Running command: meme motifs/PhyR-2.fasta -oc motifs/PhyR-2 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/PhyR-2'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 10 p0 10 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 6000 highwater mark: seq 10 pos 594\n",
      "BALANCE: samples 10 chars 6000 nodes 8 chars/node 750\n",
      "Initializing the motif probability tables for 2 to 10 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    10, min_w= 600, max_w=  600, total_size=     6000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=2\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=3\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=4\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "motif=5\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 4 motifs across 21 sites\n",
      "Finding motifs for 55 sequences\n",
      "Running command: meme motifs/Unchar-18.fasta -oc motifs/Unchar-18 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 18\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-18'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 55 p0 55 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 33000 highwater mark: seq 55 pos 594\n",
      "BALANCE: samples 55 chars 33000 nodes 8 chars/node 4125\n",
      "Initializing the motif probability tables for 2 to 55 sites...\n",
      "nsites = 48\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    55, min_w= 600, max_w=  600, total_size=    33000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 18 36 55\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  55, iter=  40 \n",
      "motif=2\n",
      "SEED DEPTHS: 18 36 55\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  55, iter=  40 \n",
      "motif=3\n",
      "SEED DEPTHS: 18 36 55\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  55, iter=  40 \n",
      "motif=4\n",
      "SEED DEPTHS: 18 36 55\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  55, iter=  40 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 3 motifs across 157 sites\n",
      "Finding motifs for 10 sequences\n",
      "Running command: meme motifs/Unchar-19.fasta -oc motifs/Unchar-19 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-19'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 10 p0 10 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 6000 highwater mark: seq 10 pos 594\n",
      "BALANCE: samples 10 chars 6000 nodes 8 chars/node 750\n",
      "Initializing the motif probability tables for 2 to 10 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    10, min_w= 600, max_w=  600, total_size=     6000\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 10\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  10, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 13 sequences\n",
      "Running command: meme motifs/Unchar-20.fasta -oc motifs/Unchar-20 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-20'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 13 p0 13 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 7800 highwater mark: seq 13 pos 594\n",
      "BALANCE: samples 13 chars 7800 nodes 8 chars/node 975\n",
      "Initializing the motif probability tables for 2 to 13 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    13, min_w= 600, max_w=  600, total_size=     7800\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 4 8 13\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  13, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Sulfur-2'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 4 p0 4 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 2400 highwater mark: seq 4 pos 594\n",
      "BALANCE: samples 4 chars 2400 nodes 8 chars/node 300\n",
      "Initializing the motif probability tables for 2 to 4 sites...\n",
      "\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     4, min_w= 600, max_w=  600, total_size=     2400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 2 4\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 4 sequences\n",
      "Running command: meme motifs/Sulfur-2.fasta -oc motifs/Sulfur-2 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 2\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "em: w=  38, psites=   2, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Unchar-21'.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 12 sequences\n",
      "Running command: meme motifs/Unchar-21.fasta -oc motifs/Unchar-21 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 12 p0 12 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 7200 highwater mark: seq 12 pos 594\n",
      "BALANCE: samples 12 chars 7200 nodes 8 chars/node 900\n",
      "Initializing the motif probability tables for 2 to 12 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    12, min_w= 600, max_w=  600, total_size=     7200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 4 8 12\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  12, iter=  10 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 12 sequences\n",
      "Running command: meme motifs/Unchar-22.fasta -oc motifs/Unchar-22 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 4\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-22'.\n",
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 12 p0 12 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 7200 highwater mark: seq 12 pos 594\n",
      "BALANCE: samples 12 chars 7200 nodes 8 chars/node 900\n",
      "Initializing the motif probability tables for 2 to 12 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=    12, min_w= 600, max_w=  600, total_size=     7200\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 4 8 12\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=  12, iter=   0 \n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n",
      "Writing results to output directory 'motifs/Unchar-23'.\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n",
      "Finding motifs for 9 sequences\n",
      "Running command: meme motifs/Unchar-23.fasta -oc motifs/Unchar-23 -dna -mod zoops -p 8 -nmotifs 5 -evt 0.001 -minw 6 -maxw 40 -allw -minsites 3\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "BACKGROUND: using background model of order 0\n",
      "PRIMARY (classic): n 9 p0 9 p1 0 p2 0\n",
      "SEQUENCE GROUP USAGE-- Starts/EM: p0; Trim: p0; pvalue: p0; nsites: p0,p1,p2\n",
      "SEEDS: maxwords 5400 highwater mark: seq 9 pos 594\n",
      "BALANCE: samples 9 chars 5400 nodes 8 chars/node 675\n",
      "Initializing the motif probability tables for 2 to 9 sites...\n",
      "nsites = 8\n",
      "Done initializing.\n",
      "all widths from min to max\n",
      "\n",
      "seqs=     9, min_w= 600, max_w=  600, total_size=     5400\n",
      "\n",
      "motif=1\n",
      "SEED DEPTHS: 3 6 9\n",
      "SEED WIDTHS: 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40\n",
      "em: w=  40, psites=   9, iter=   0 \n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No motif found with E-value < 1.0e-03\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/motif.py:311: FutureWarning: The default dtype for empty Series will be 'object' instead of 'float64' in a future version. Specify a dtype explicitly to silence this warning.\n",
      "  idx1 = pd.Series([], name=\"motif\")\n"
     ]
    }
   ],
   "source": [
    "for k in ica_data.imodulon_names:\n",
    "    find_motifs(ica_data, k, fasta_file, force=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Compare motifs to known databases using TOMTOM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-1/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-1/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-1/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.91077\n",
      "#   Estimated pi_0=0.913018\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.943845\n",
      "#   Estimated pi_0=0.947375\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.949673\n",
      "#   Estimated pi_0=0.950482\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938476\n",
      "#   Estimated pi_0=0.938476\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/T3SS/tomtom -thresh 0.001 -incomplete-scores -png motifs/T3SS/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/T3SS/tomtom'.\n",
      "Processing query 1 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961749\n",
      "#   Estimated pi_0=0.962025\n",
      "Processing query 2 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998909\n",
      "#   Estimated pi_0=0.999923\n",
      "Processing query 3 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.981565\n",
      "#   Estimated pi_0=0.983667\n",
      "Processing query 4 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961795\n",
      "#   Estimated pi_0=0.966636\n",
      "Processing query 5 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.970048\n",
      "#   Estimated pi_0=0.973791\n",
      "Processing query 6 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.98383\n",
      "#   Estimated pi_0=0.98383\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/FixK2-1/tomtom -thresh 0.001 -incomplete-scores -png motifs/FixK2-1/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/FixK2-1/tomtom'.\n",
      "Processing query 1 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.979517\n",
      "#   Estimated pi_0=0.979517\n",
      "Processing query 2 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.91077\n",
      "#   Estimated pi_0=0.91077\n",
      "Processing query 3 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.912119\n",
      "#   Estimated pi_0=0.912654\n",
      "Processing query 4 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.927926\n",
      "#   Estimated pi_0=0.930626\n",
      "Processing query 5 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.935498\n",
      "#   Estimated pi_0=0.935859\n",
      "Processing query 6 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.912273\n",
      "#   Estimated pi_0=0.913001\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/bll2166/tomtom -thresh 0.001 -incomplete-scores -png motifs/bll2166/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/bll2166/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.941032\n",
      "#   Estimated pi_0=0.943575\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.926681\n",
      "#   Estimated pi_0=0.927173\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.924986\n",
      "#   Estimated pi_0=0.929338\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/NifA/tomtom -thresh 0.001 -incomplete-scores -png motifs/NifA/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/NifA/tomtom'.\n",
      "Processing query 1 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.949059\n",
      "#   Estimated pi_0=0.955302\n",
      "Processing query 2 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.911924\n",
      "#   Estimated pi_0=0.912119\n",
      "Processing query 3 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.930626\n",
      "#   Estimated pi_0=0.930626\n",
      "Processing query 4 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921646\n",
      "#   Estimated pi_0=0.928845\n",
      "Processing query 5 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.930178\n",
      "#   Estimated pi_0=0.93237\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/PhyR-1/tomtom -thresh 0.001 -incomplete-scores -png motifs/PhyR-1/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/PhyR-1/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.908594\n",
      "#   Estimated pi_0=0.908594\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.904492\n",
      "#   Estimated pi_0=0.906832\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.935859\n",
      "#   Estimated pi_0=0.936669\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.92664\n",
      "#   Estimated pi_0=0.927589\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Denitrification/tomtom -thresh 0.001 -incomplete-scores -png motifs/Denitrification/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Denitrification/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999923\n",
      "#   Estimated pi_0=0.999923\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998137\n",
      "#   Estimated pi_0=0.999073\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997373\n",
      "#   Estimated pi_0=0.997373\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Irr/tomtom -thresh 0.001 -incomplete-scores -png motifs/Irr/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Irr/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938476\n",
      "#   Estimated pi_0=0.939737\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.951777\n",
      "#   Estimated pi_0=0.956452\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917781\n",
      "#   Estimated pi_0=0.920749\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.945501\n",
      "#   Estimated pi_0=0.947015\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Multiple_stress/tomtom -thresh 0.001 -incomplete-scores -png motifs/Multiple_stress/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Multiple_stress/tomtom'.\n",
      "Processing query 1 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.942837\n",
      "#   Estimated pi_0=0.945657\n",
      "Processing query 2 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.96764\n",
      "#   Estimated pi_0=0.96764\n",
      "Processing query 3 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.995533\n",
      "#   Estimated pi_0=0.997099\n",
      "Processing query 4 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.982451\n",
      "#   Estimated pi_0=0.98383\n",
      "Processing query 5 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.957932\n",
      "#   Estimated pi_0=0.960303\n",
      "Processing query 6 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933243\n",
      "#   Estimated pi_0=0.933243\n",
      "Processing query 7 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973151\n",
      "#   Estimated pi_0=0.973373\n",
      "Processing query 8 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.962568\n",
      "#   Estimated pi_0=0.962568\n",
      "Processing query 9 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.95289\n",
      "#   Estimated pi_0=0.957091\n",
      "Processing query 10 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.97753\n",
      "#   Estimated pi_0=0.977724\n",
      "Processing query 11 out of 11 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00134\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/bll3348/tomtom -thresh 0.001 -incomplete-scores -png motifs/bll3348/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/bll3348/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997373\n",
      "#   Estimated pi_0=0.997373\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00332\n",
      "#   Estimated pi_0=1\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.994403\n",
      "#   Estimated pi_0=0.995573\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/XylR_bll3916/tomtom -thresh 0.001 -incomplete-scores -png motifs/XylR_bll3916/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/XylR_bll3916/tomtom'.\n",
      "Processing query 1 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917037\n",
      "#   Estimated pi_0=0.918415\n",
      "Processing query 2 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.91268\n",
      "#   Estimated pi_0=0.91268\n",
      "Processing query 3 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.914763\n",
      "#   Estimated pi_0=0.914763\n",
      "Processing query 4 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.908362\n",
      "#   Estimated pi_0=0.911238\n",
      "Processing query 5 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.929297\n",
      "#   Estimated pi_0=0.929754\n",
      "Processing query 6 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.908594\n",
      "#   Estimated pi_0=0.910023\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-6/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-6/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-6/tomtom'.\n",
      "Processing query 1 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.902425\n",
      "#   Estimated pi_0=0.902425\n",
      "Processing query 2 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.983667\n",
      "#   Estimated pi_0=0.984858\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-7/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-7/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-7/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.924118\n",
      "#   Estimated pi_0=0.926761\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.923576\n",
      "#   Estimated pi_0=0.92762\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917176\n",
      "#   Estimated pi_0=0.919517\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.946656\n",
      "#   Estimated pi_0=0.946679\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/HrcA/tomtom -thresh 0.001 -incomplete-scores -png motifs/HrcA/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/HrcA/tomtom'.\n",
      "Processing query 1 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.918912\n",
      "#   Estimated pi_0=0.918912\n",
      "Processing query 2 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.90446\n",
      "#   Estimated pi_0=0.90446\n",
      "Processing query 3 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.90446\n",
      "#   Estimated pi_0=0.90446\n",
      "Processing query 4 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.90827\n",
      "#   Estimated pi_0=0.911438\n",
      "Processing query 5 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00247\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Signal_transduction/tomtom -thresh 0.001 -incomplete-scores -png motifs/Signal_transduction/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Signal_transduction/tomtom'.\n",
      "Processing query 1 out of 1 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00247\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Leucine/tomtom -thresh 0.001 -incomplete-scores -png motifs/Leucine/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Leucine/tomtom'.\n",
      "Processing query 1 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.969002\n",
      "#   Estimated pi_0=0.969002\n",
      "Processing query 2 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.968276\n",
      "#   Estimated pi_0=0.96964\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Flagella/tomtom -thresh 0.001 -incomplete-scores -png motifs/Flagella/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Flagella/tomtom'.\n",
      "Processing query 1 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.929716\n",
      "#   Estimated pi_0=0.932795\n",
      "Processing query 2 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997099\n",
      "#   Estimated pi_0=0.998992\n",
      "Processing query 3 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.938786\n",
      "#   Estimated pi_0=0.938786\n",
      "Processing query 4 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.975571\n",
      "#   Estimated pi_0=0.975777\n",
      "Processing query 5 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997191\n",
      "#   Estimated pi_0=0.997373\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Translation-1/tomtom -thresh 0.001 -incomplete-scores -png motifs/Translation-1/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Translation-1/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.92325\n",
      "#   Estimated pi_0=0.92631\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913706\n",
      "#   Estimated pi_0=0.913706\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.967014\n",
      "#   Estimated pi_0=0.971619\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-8/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-8/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-8/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913706\n",
      "#   Estimated pi_0=0.916953\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961749\n",
      "#   Estimated pi_0=0.962897\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973809\n",
      "#   Estimated pi_0=0.9749\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Oxidation_stress/tomtom -thresh 0.001 -incomplete-scores -png motifs/Oxidation_stress/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Oxidation_stress/tomtom'.\n",
      "Processing query 1 out of 1 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.92016\n",
      "#   Estimated pi_0=0.92016\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-9/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-9/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-9/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.915799\n",
      "#   Estimated pi_0=0.916813\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921515\n",
      "#   Estimated pi_0=0.921993\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00503\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Phenylalanine/tomtom -thresh 0.001 -incomplete-scores -png motifs/Phenylalanine/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Phenylalanine/tomtom'.\n",
      "Processing query 1 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961153\n",
      "#   Estimated pi_0=0.963165\n",
      "Processing query 2 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00162\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/FixK2-2/tomtom -thresh 0.001 -incomplete-scores -png motifs/FixK2-2/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/FixK2-2/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.967989\n",
      "#   Estimated pi_0=0.967989\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.911238\n",
      "#   Estimated pi_0=0.913877\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.905028\n",
      "#   Estimated pi_0=0.905028\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.909147\n",
      "#   Estimated pi_0=0.909583\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-10/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-10/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-10/tomtom'.\n",
      "Processing query 1 out of 1 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.964901\n",
      "#   Estimated pi_0=0.966884\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-11/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-11/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-11/tomtom'.\n",
      "Processing query 1 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.92016\n",
      "#   Estimated pi_0=0.92325\n",
      "Processing query 2 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944503\n",
      "#   Estimated pi_0=0.945765\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/HutC/tomtom -thresh 0.001 -incomplete-scores -png motifs/HutC/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/HutC/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00247\n",
      "#   Estimated pi_0=1\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.998992\n",
      "#   Estimated pi_0=0.998992\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00247\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-12/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-12/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-12/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.936994\n",
      "#   Estimated pi_0=0.936994\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961556\n",
      "#   Estimated pi_0=0.965412\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.940026\n",
      "#   Estimated pi_0=0.940026\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-14/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-14/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-14/tomtom'.\n",
      "Processing query 1 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.922856\n",
      "#   Estimated pi_0=0.926353\n",
      "Processing query 2 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.905591\n",
      "#   Estimated pi_0=0.905591\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/AA_transport/tomtom -thresh 0.001 -incomplete-scores -png motifs/AA_transport/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/AA_transport/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.960589\n",
      "#   Estimated pi_0=0.962298\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.981387\n",
      "#   Estimated pi_0=0.983287\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997006\n",
      "#   Estimated pi_0=0.998223\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.995271\n",
      "#   Estimated pi_0=0.997191\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Glyoxylate/tomtom -thresh 0.001 -incomplete-scores -png motifs/Glyoxylate/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Glyoxylate/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.918415\n",
      "#   Estimated pi_0=0.918912\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.961227\n",
      "#   Estimated pi_0=0.963503\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.910483\n",
      "#   Estimated pi_0=0.9109\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/NnrR/tomtom -thresh 0.001 -incomplete-scores -png motifs/NnrR/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/NnrR/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.966505\n",
      "#   Estimated pi_0=0.969474\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.997228\n",
      "#   Estimated pi_0=0.999923\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.976181\n",
      "#   Estimated pi_0=0.978107\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.97897\n",
      "#   Estimated pi_0=0.981387\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Cell_mobility/tomtom -thresh 0.001 -incomplete-scores -png motifs/Cell_mobility/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Cell_mobility/tomtom'.\n",
      "Processing query 1 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.908594\n",
      "#   Estimated pi_0=0.908594\n",
      "Processing query 2 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.894866\n",
      "#   Estimated pi_0=0.895231\n",
      "Processing query 3 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913706\n",
      "#   Estimated pi_0=0.916454\n",
      "Processing query 4 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.924927\n",
      "#   Estimated pi_0=0.928899\n",
      "Processing query 5 out of 5 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.922776\n",
      "#   Estimated pi_0=0.922776\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/T6SS/tomtom -thresh 0.001 -incomplete-scores -png motifs/T6SS/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/T6SS/tomtom'.\n",
      "Processing query 1 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00142\n",
      "#   Estimated pi_0=1\n",
      "Processing query 2 out of 2 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00247\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Sulfur-1/tomtom -thresh 0.001 -incomplete-scores -png motifs/Sulfur-1/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Sulfur-1/tomtom'.\n",
      "Processing query 1 out of 7 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.956183\n",
      "#   Estimated pi_0=0.959681\n",
      "Processing query 2 out of 7 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944503\n",
      "#   Estimated pi_0=0.948437\n",
      "Processing query 3 out of 7 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.971327\n",
      "#   Estimated pi_0=0.973373\n",
      "Processing query 4 out of 7 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.929696\n",
      "#   Estimated pi_0=0.931088\n",
      "Processing query 5 out of 7 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.996818\n",
      "#   Estimated pi_0=0.998992\n",
      "Processing query 6 out of 7 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00156\n",
      "#   Estimated pi_0=1\n",
      "Processing query 7 out of 7 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.936926\n",
      "#   Estimated pi_0=0.936926\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-16/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-16/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-16/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933695\n",
      "#   Estimated pi_0=0.934987\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.944215\n",
      "#   Estimated pi_0=0.944215\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.951999\n",
      "#   Estimated pi_0=0.953621\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00247\n",
      "#   Estimated pi_0=1\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Copper/tomtom -thresh 0.001 -incomplete-scores -png motifs/Copper/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Copper/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.954736\n",
      "#   Estimated pi_0=0.95592\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.98415\n",
      "#   Estimated pi_0=0.98415\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00055\n",
      "#   Estimated pi_0=1\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.995573\n",
      "#   Estimated pi_0=0.995573\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-17/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-17/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-17/tomtom'.\n",
      "Processing query 1 out of 1 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.931078\n",
      "#   Estimated pi_0=0.931955\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Bacterial_chemotaxis/tomtom -thresh 0.001 -incomplete-scores -png motifs/Bacterial_chemotaxis/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Bacterial_chemotaxis/tomtom'.\n",
      "Processing query 1 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.920647\n",
      "#   Estimated pi_0=0.921993\n",
      "Processing query 2 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.913882\n",
      "#   Estimated pi_0=0.913882\n",
      "Processing query 3 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.884317\n",
      "#   Estimated pi_0=0.886806\n",
      "Processing query 4 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.91231\n",
      "#   Estimated pi_0=0.91231\n",
      "Processing query 5 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.917914\n",
      "#   Estimated pi_0=0.919403\n",
      "Processing query 6 out of 6 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.948252\n",
      "#   Estimated pi_0=0.950686\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/PhyR-2/tomtom -thresh 0.001 -incomplete-scores -png motifs/PhyR-2/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/PhyR-2/tomtom'.\n",
      "Processing query 1 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.999923\n",
      "#   Estimated pi_0=0.999923\n",
      "Processing query 2 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00503\n",
      "#   Estimated pi_0=1\n",
      "Processing query 3 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 1.00417\n",
      "#   Estimated pi_0=1\n",
      "Processing query 4 out of 4 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.973606\n",
      "#   Estimated pi_0=0.976578\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Running command: tomtom -oc motifs/Unchar-18/tomtom -thresh 0.001 -incomplete-scores -png motifs/Unchar-18/meme.txt /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/collectf.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/dpinteract.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/prodoric.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/regtransbase.meme /home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/pymodulon/data/meme/SwissRegulon_e_coli.meme\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Writing results to output directory 'motifs/Unchar-18/tomtom'.\n",
      "Processing query 1 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.925754\n",
      "#   Estimated pi_0=0.928479\n",
      "Processing query 2 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.933727\n",
      "#   Estimated pi_0=0.935398\n",
      "Processing query 3 out of 3 \n",
      "# Computing q-values.\n",
      "#   Estimating pi_0 from all 1182 observed p-values.\n",
      "#   Estimating pi_0.\n",
      "# Minimal pi_zero = 0.921904\n",
      "#   Estimated pi_0=0.926761\n",
      "/home/gzp/anaconda3/envs/pymodulon/lib/python3.10/site-packages/bs4/builder/__init__.py:545: XMLParsedAsHTMLWarning: It looks like you're parsing an XML document using an HTML parser. If this really is an HTML document (maybe it's XHTML?), you can ignore or filter this warning. If it's XML, you should know that using an XML parser will be more reliable. To parse this document as XML, make sure you have the lxml package installed, and pass the keyword argument `features=\"xml\"` into the BeautifulSoup constructor.\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "for info in ica_data.motif_info.values():\n",
    "    compare_motifs(info)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Unchar-1\n",
      "FixK2-1\n",
      "bll2166\n",
      "NifA\n",
      "PhyR-1\n",
      "Irr\n",
      "Multiple_stress\n",
      "XylR/bll3916\n",
      "Unchar-6\n",
      "Unchar-7\n",
      "HrcA\n",
      "Flagella\n",
      "Translation-1\n",
      "Unchar-8\n",
      "Oxidation_stress\n",
      "Unchar-9\n",
      "FixK2-2\n",
      "Unchar-11\n",
      "Unchar-14\n",
      "Glyoxylate\n",
      "NnrR\n",
      "Cell_mobility\n",
      "Unchar-16\n",
      "Bacterial_chemotaxis\n",
      "Unchar-18\n"
     ]
    }
   ],
   "source": [
    "for k in ica_data.motif_info.keys():\n",
    "    if not ica_data.motif_info[k].matches.empty:\n",
    "        print(k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from os import path\n",
    "save_to_json(ica_data, path.join('..','data','interim','bd_raw.json.gz'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
